File: node/src/file-manager.ts
import fs from 'fs';

export class FileManager {
    private readonly rutaArchivo: string;

    constructor(nombreArchivo: string) {
        this.rutaArchivo = `${nombreArchivo}`;
    }

    private esEscribible(): boolean {
        try {
            fs.accessSync(this.rutaArchivo, fs.constants.W_OK);
            return true;
        } catch (err) {
            return false;
        }
    }

    apppend(data: string): void {

        if (!fs.existsSync(this.rutaArchivo) || !this.esEscribible()) {
            throw new Error(`La ruta del archivo '${this.rutaArchivo}' no existe o no es escribible.`);
        }
        try {
            fs.writeFileSync(this.rutaArchivo, data, { flag: 'a' });
        } catch (err) {
            console.error('Error al guardar el archivo:', err);
        }
    }
}

File: golang/utils/file-manager.go
package utils

import (
	"fmt"
	"log"
	"os"
)

// CLASS
type FileManager struct {
	rutaArchivo string
}

// CONSTRUCTOR
func NewFileManager(nombreArchivo string) (*FileManager, error) {
	rutaArchivo := nombreArchivo
	if _, err := os.Stat(rutaArchivo); os.IsNotExist(err) {
		return nil, fmt.Errorf("el archivo %s no existe", rutaArchivo)
	}
	return &FileManager{rutaArchivo: rutaArchivo}, nil
}

// method
func (fm *FileManager) Apppend(data string) {

	file, err := os.OpenFile(fm.rutaArchivo, os.O_CREATE|os.O_APPEND|os.O_WRONLY, 0644)
	if err != nil {
		log.Fatal(err)
	}
	defer file.Close()

	_, err = file.Write([]byte(data))
	if err != nil {
		log.Fatal(err)
	}

	if err != nil {
		log.Fatal(err)
	}

}
